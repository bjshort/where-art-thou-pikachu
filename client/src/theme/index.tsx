import styled, { createGlobalStyle } from 'styled-components';
import reset from 'styled-reset';
export const Theme = {
  colors: {
    body: '#333333',
    body2: '#666666',
    primaryBrand: '#0072df',
    secondaryBrand: '#003594',
    tertiaryBrand: '#00abf1',
    placeholder: '#aaaaaa',
    disabled: '#cccccc',
    inactive: '#dfe4ed',
    positive: '#5fce06',
    negative: '#f03838',
    warning: '#fa7200',
    bgAccent: '#f5f7fb',
    dividerLight: '#eeeeee',
    dividerDark: '#dddddd',
  },
  typography: {
    face: {
      primary: `"Poppins", "Helvetica Neue", sans-serif`,
      secondary: `"SourceSansPro", sans-serif`,
    },
  },
};

const breakpoints: { [key: string]: number } = {
  sm: 576,
  md: 768,
  lg: 1200,
};

export const getBreakpoint = (breakpoint: string): string => {
  return `${breakpoints[breakpoint]}px`;
};

export const GlobalStyle = createGlobalStyle`
  ${reset}

  html, body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    overflow-x: hidden;
  }

  body {
    background: #f5f7fb;
    font-size: 16px;
    font-family: ${Theme.typography.face.primary};

    @media screen and (max-width: ${getBreakpoint('md')}) {
      font-size: 14px;
    }

    @media screen and (max-width: ${getBreakpoint('sm')}) {
      font-size: 13px;
    }
  }
`;

export const Header1 = styled.h1`
  color: ${(props) => (props.color ? props.color : props.theme.colors.body)};
  font-family: ${(props) => props.theme.typography.face.primary};
  font-size: 3em;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.25;
  letter-spacing: 0.006em;
  margin-bottom: 1rem;
`;

export const Header2 = styled.h2`
  color: ${(props) => (props.color ? props.color : props.theme.colors.body)};
  font-family: ${(props) => props.theme.typography.face.primary};
  font-size: 2.25em;
  font-weight: 700;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.33;
  letter-spacing: 0.006em;
  margin-bottom: 1rem;
`;

export const Header3 = styled.h3`
  color: ${(props) => (props.color ? props.color : props.theme.colors.body)};
  font-family: ${(props) => props.theme.typography.face.primary};
  font-size: 1.25em;
  font-weight: 700;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.8;
  letter-spacing: 0.006em;
  margin-bottom: 1rem;
`;

export const Header4 = styled.h4`
  color: ${(props) => (props.color ? props.color : props.theme.colors.body)};
  font-family: ${(props) => props.theme.typography.face.secondary};
  font-size: 1em;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.5;
  letter-spacing: 0.013em;
  margin-bottom: 1rem;
`;

export const Header5 = styled.h5`
  color: ${(props) => (props.color ? props.color : props.theme.colors.body)};
  font-family: ${(props) => props.theme.typography.face.secondary};
  font-size: 0.875em;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.5;
  letter-spacing: 0.013em;
  margin-bottom: 1rem;
`;
