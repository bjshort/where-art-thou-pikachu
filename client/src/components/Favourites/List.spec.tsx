import * as React from 'react';
import { ReactWrapper, mount, shallow } from 'enzyme';
import FavouritesList from './List';
import renderer from 'react-test-renderer';
// import 'jest-enzyme';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { Theme } from '../../theme';

let component: ReactWrapper;
const mockFavourites = [
  {
    id: 54,
    name: 'psyduck',
    description: {
      original: 'Quack',
      shakespeare: 'Quack',
    },
    imageUrl: 'https://com.image.png',
    height: '2.62',
    weight: '43.21',
    exp: 64,
  },
  {
    id: 143,
    name: 'snorlax',
    description: {
      original: 'zzz',
      shakespeare: 'zzz',
    },
    imageUrl: 'https://com.image.png',
    height: '6.89',
    weight: '1014.11',
    exp: 189,
  },
];

describe('FavouritesList', () => {
  it('should render', () => {
    const component = (
      <ThemeProvider theme={Theme}>
        <FavouritesList items={mockFavourites} />{' '}
      </ThemeProvider>
    );

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
