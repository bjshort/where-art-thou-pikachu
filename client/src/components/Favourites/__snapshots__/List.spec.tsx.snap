// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesList should render 1`] = `
Array [
  .c0 {
  color: #333333;
  font-family: "Poppins","Helvetica Neue",sans-serif;
  font-size: 2.25em;
  font-weight: 700;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.33;
  -webkit-letter-spacing: 0.006em;
  -moz-letter-spacing: 0.006em;
  -ms-letter-spacing: 0.006em;
  letter-spacing: 0.006em;
  margin-bottom: 1rem;
}

.c1 {
  font-family: 'Press Start 2P',cursive;
  background-color: #000000;
  padding: 5px;
  border: 2px solid #fff;
  color: #ffffff;
  margin-bottom: 7px;
}

.c3 {
  background-color: #ffffff;
  width: 100px;
}

.c5 {
  display: inline-block;
  vertical-align: top;
  margin-left: 15px;
}

.c6 {
  text-transform: uppercase;
  color: #ffffff;
  display: block;
  font-size: 12px;
  line-height: 25px;
}

.c8 {
  display: block;
  font-size: 12px;
  line-height: 25px;
  -webkit-letter-spacing: 0.4em;
  -moz-letter-spacing: 0.4em;
  -ms-letter-spacing: 0.4em;
  letter-spacing: 0.4em;
}

.c2 {
  display: inline-block;
}

.c4 {
  display: block;
  font-size: 11px;
  text-align: center;
  line-height: 20px;
}

.c7 {
  border: 2px solid #fb8c4b;
}

<div>
    <h2
      className="c0"
    >
      Your favourites
    </h2>
    <ul
      className=""
    >
      <li
        className=""
      >
        <div
          className="c1"
        >
          <div
            className="c2"
          >
            <img
              alt="psyduck"
              className="c3"
              src="https://com.image.png"
            />
            <p
              className="c4"
            >
              No. 
              54
            </p>
          </div>
          <div
            className="c5"
          >
            <p
              className="c6"
            >
              psyduck
            </p>
            <p
              className="c6"
            >
              HT 
              2.62
            </p>
            <p
              className="c6"
            >
              WT 
              43.21
              lb
            </p>
            <p
              className="c6"
            >
              EXP 
              64
            </p>
          </div>
          <hr
            className="c7"
          />
          <p
            className="c8"
          >
            Quack
          </p>
        </div>
      </li>
      <li
        className=""
      >
        <div
          className="c1"
        >
          <div
            className="c2"
          >
            <img
              alt="snorlax"
              className="c3"
              src="https://com.image.png"
            />
            <p
              className="c4"
            >
              No. 
              143
            </p>
          </div>
          <div
            className="c5"
          >
            <p
              className="c6"
            >
              snorlax
            </p>
            <p
              className="c6"
            >
              HT 
              6.89
            </p>
            <p
              className="c6"
            >
              WT 
              1014.11
              lb
            </p>
            <p
              className="c6"
            >
              EXP 
              189
            </p>
          </div>
          <hr
            className="c7"
          />
          <p
            className="c8"
          >
            zzz
          </p>
        </div>
      </li>
    </ul>
  </div>,
  " ",
]
`;
